framework/include/materials/MaterialWarehouse.h

```diff
@@ -49,17 +49,17 @@ class MaterialWarehouse : public MooseObjectWarehouse<Material>
   /**
    * A special method unique to this class for adding Block, Neighbor, and Face material objects.
    */
-  void addObjects(std::shared_ptr<Material> block,
-                  std::shared_ptr<Material> neighbor,
-                  std::shared_ptr<Material> face,
+  void addObjects(std::shared_ptr<MaterialBase> block,
+                  std::shared_ptr<MaterialBase> neighbor,
+                  std::shared_ptr<MaterialBase> face,
```

Comment:
I would not change this. Just use this to add the `Material`s (see my comments on `bnd` and `neighbor`).

Add a new method for `InterfaceMaterials`